nox/sessions.py

```diff
@@ -210,10 +212,37 @@ def invoked_from(self) -> str:
         """
         return self._runner.global_config.invoked_from
 
-    def chdir(self, dir: Union[str, os.PathLike]) -> None:
-        """Change the current working directory."""
+    class _WorkingDirContext:
```

Comment:
I really dislike nested classes and functions. The main reason to do it is for capture, which is a bad, unreadable reason. Would this be better as an un-nested class?

Reply:
Certainly not. It is one of my bad habits. I've moved it out in the global scope 👍